cloudinary-core/src/main/java/com/cloudinary/Url.java

```diff
@@ -417,6 +419,16 @@ public String generate(String source) {
             } catch (MalformedURLException ignored) {
             }
         }
+        if (cloudinary.analytics != null && cloudinary.config.analytics) {
```

Comment:
I'm not sure this line is clear enough.

What's the difference between `cloudinary.analytics` and `cloudinary.config.analytics`?
Why do we both conditions need to be true?

Reply:
This was part of the design by @const-cloudinary and I
Think of the situation where we set the analytics object by default from the Android by the developer explicitly does not want to send analytics
@const-cloudinary suggest we do this as part of the config we init, so if the develop is running it through some kind of YML, XML or any other config file he can just set analytics = false

Reply:
`cloudinary.config.analytics` is the configuration(boolean) for turning analytics on and off. `cloudinary.analytics` is the Analytics class itself that generates the token. It can be configured from the outside (android) with the relevant SDK key and other things.

Reply:
And I agree with @patrick-tolosa, that class should be always initialized in the constructor (whether in Java SDK or in Android)

Reply:
The condition seems a bit off for anyone not in the details.
a flag of "on and off" is already called `analytics` in other SDKs (A bit unfortunate), but the perhaps we can find a better name for the class/instance?

Reply:
So according to what I read, we should init the analytics in the constructor (with Java params)
And we want to change hte name of the class `analytics`? Let me know what name you want to put instead